### ISSUE TITLE
Cannot launch experiments for some datasets (PATTERN/CLUSTER/Peptides-func/Peptides-struct)

### ISSUE THREAD

**Issue by wsjeon** [rampasek/GraphGPS#25]
Hi, thanks for your nice work.

I tried running GraphGPS for all datasets, but I could find out the following errors for the datasets below:

- PATTERN (from configs/GPS/pattern-GPS.yaml) / CLUSTER (from configs/GPS/cluster-GPS.yaml)
```
Traceback (most recent call last):
  File "/usr2/wjeon/scratch/GraphGPS/main.py", line 143, in <module>
    loaders = create_loader()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 311, in create_loader
    dataset = create_dataset()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 242, in create_dataset
    dataset = load_dataset()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 187, in load_dataset
    dataset = func(format, name, dataset_dir)
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/master_loader.py", line 110, in load_dataset_master
    dataset = preformat_GNNBenchmarkDataset(dataset_dir, name)
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/master_loader.py", line 287, in preformat_GNNBenchmarkDataset
    return dataset
UnboundLocalError: local variable 'dataset' referenced before assignment
```
- peptides-func (from configs/GPS/peptides-func) / peptides-struct (from configs/GPS/peptides-struct)
```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/opt/conda/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/conda/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/opt/conda/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/conda/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/opt/conda/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/opt/conda/lib/python3.10/http/client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/opt/conda/lib/python3.10/ssl.py", line 513, in wrap_socket
    return self.sslsocket_class._create(
  File "/opt/conda/lib/python3.10/ssl.py", line 1071, in _create
    self.do_handshake()
  File "/opt/conda/lib/python3.10/ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr2/wjeon/scratch/GraphGPS/main.py", line 142, in <module>
    loaders = create_loader()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 312, in create_loader
    dataset = create_dataset()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 243, in create_dataset
    dataset = load_dataset()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/graphgym/loader.py", line 188, in load_dataset
    dataset = func(format, name, dataset_dir)
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/master_loader.py", line 160, in load_dataset_master
    dataset = preformat_Peptides(dataset_dir, name)
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/master_loader.py", line 522, in preformat_Peptides
    dataset = PeptidesFunctionalDataset(dataset_dir)
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/dataset/peptides_functional.py", line 56, in __init__
    super().__init__(self.folder, transform, pre_transform)
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/data/in_memory_dataset.py", line 55, in __init__
    super().__init__(root, transform, pre_transform, pre_filter, log)
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/data/dataset.py", line 91, in __init__
    self._download()
  File "/opt/conda/lib/python3.10/site-packages/torch_geometric/data/dataset.py", line 185, in _download
    self.download()
  File "/local/mnt/workspace/scratch/wjeon/GraphGPS/graphgps/loader/dataset/peptides_functional.py", line 75, in download
    if decide_download(self.url):
  File "/opt/conda/lib/python3.10/site-packages/ogb/utils/url.py", line 12, in decide_download
    d = ur.urlopen(url)
  File "/opt/conda/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/conda/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/opt/conda/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/opt/conda/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/opt/conda/lib/python3.10/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/opt/conda/lib/python3.10/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)>
```

May I ask how we can launch the code for those datasets?

Thanks!

**Comment by wsjeon** [comment]
For `peptides-func` and `peptides-struct`, I could resolve issues by following [this link](https://stackoverflow.com/a/57910133).

**Comment by rampasek** [comment]
Hi!
Thanks for reporting the issue!
It was indeed a bug from accumulated edits to `preformat_GNNBenchmarkDataset()` function, it is now fixed by PR #26
Please let me know if there are still any other issues!

**Comment by wsjeon** [comment]
Thanks for resolving and closing this issue! Everything works well on my end 👍
